src/ViewModels/ShellViewModel.cs

```diff
@@ -54,7 +55,13 @@ public INavigationViewService NavigationViewService
 
     public async Task OnLoaded()
     {
-        if (await _localSettingsService.ReadSettingAsync<bool>(WellKnownSettingsKeys.IsNotFirstRun))
+        // Load the Machine Configuration Page if the app is launched via file activation
+        var activatedEventArgs = AppInstance.GetCurrent().GetActivatedEventArgs();
+        if (activatedEventArgs.Kind == ExtendedActivationKind.File)
+        {
+            NavigationService.NavigateTo(typeof(DevHome.SetupFlow.ViewModels.SetupFlowViewModel).FullName!);

```

Comment:
NavigateTo is called in the file handler class, is this line still needed?

Reply:
If you remove that line the behavior I was seeing was that on launch you'd still end up on the dashboard. I was thinking this may be the case if this is called after the activation handling kicks in